python/pyspark/sql/worker/plan_data_source_read.py

```diff
@@ -258,8 +261,25 @@ def batched(iterator: Iterator, n: int) -> Iterator:
                             },
                         )
 
-                    for col in range(num_cols):
-                        pylist[col].append(column_converters[col](result[col]))
+                    # Assign output values by name of the field, not position, if the result is a
+                    # named `Row` object.
+                    if isinstance(result, Row) and hasattr(result, "__fields__"):
```

Comment:
Can we match the implementation with python worker? See `assign_cols_by_name` at `worker.py`.

Reply:
This is actually different from `assign_cols_by_name` which re-arranges arrow batch columns by the arrow type names. Here we want to match a single named `Row` object to the return schema. The only way to tell whether it's named `Row(a=1, b=1)` from an unnamed `Row(1,2)` is by checking this `__fields__`.